pyinteraph/path_analysis.py

```diff
@@ -399,179 +410,219 @@ def main():
-                                             target = target_list,
-                                             maxl = args.maxl)
-            path_type = "all"
+    # get graph nodes and edges
+    nodes = graph.nodes()
+    edges = graph.edges()
+    # print nodes
+    print(f"Graph loaded! {len(nodes)} nodes, {len(edges)} edges")
+    print("Node list:")
+    print(np.array(identifiers))
+
+    ############################## PATHS ##############################
+
+    if args.do_path:
+        # Check if source and target are provided
+        if args.source and args.target:
+            # Convert user input to a list of nodes
+            source_list = convert_input_to_list(user_input = args.source,
+                                                identifiers = identifiers)
+            target_list = convert_input_to_list(user_input = args.target,
+                                                identifiers = identifiers)
         else:
+            log.error("Source and target must be provided when calculating paths.")
+            exit(1)
+
+        # Choost path type
+        if args.path_l == "shortest":
             all_paths = get_shortest_paths(graph = graph,
                                            source = source_list,
-                                           target = target_list,
-                                           maxl = args.maxl)
-            path_type = "shortest"
+                                           target = target_list)
+        else:
+            # Check integer is provided
+            try:
+                maxl = int(args.path_l)
+            except Exception:
+                errstr = "Integer must be specified for calculation of all " \
+                         "simple paths. Otherwise use 'shortest'"
+                raise ValueError(errstr)
+            # Calculate all simple paths:
+            all_paths = get_all_simple_paths(graph = graph,
+                                             source = source_list,
+                                             target = target_list,
+                                             maxl = maxl)
+            # Change output file name if user forgets to change
+            if args.p_out == "shortest_paths":
+                args.p_out = f"all_paths_{maxl}" 
         
+        # Get persistence graph
+        pers_graph = get_persistence_graph(graph = graph, 
+                                           paths = all_paths, 
+                                           identifiers = identifiers)
         # Create sorted table from paths
         all_paths_table = sort_paths(graph = graph,
                                      paths = all_paths,
                                      sort_by = args.sort_by)
-        all_paths_graph = get_graph_from_paths(all_paths)
-
 
         # Save all/shortest path table
-        write_table(f"{path_type}_{args.output}.txt", all_paths_table)
+        print(f"Saving output: {args.p_out}")
```

Comment:
Better to use `sys.stdout.write()` than `print()` if you want to print to the standard output.